### PR TITLE
[WIP] Support CheXNet and NIHChestXRays dataset

### PR DESCRIPTION
**Pull request by okotaku**
## Motivation

paper: https://arxiv.org/abs/1711.05225

## Modification

- CheXNet config
- NIHChestXRays dataset
- TenCropTestTimeAug
- ROCAUC

## Checklist

**Before PR**:

- [ ] Pre-commit or other linting tools are used to fix the potential lint issues.
- [ ] Bug fixes are fully covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, like docstring or example tutorials.

**After PR**:

- [ ] If the modification has potential influence on downstream or other related projects, this PR should be tested with those projects, like MMDet or MMSeg.
- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: configs/_base_/datasets/nih_chestxrays_bs16.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+test_dataloader = dict(
+    batch_size=16,
+    num_workers=4,
+    dataset=dict(
+        type=dataset_type,
+        data_root='data/VOCdevkit/VOC2007',
+        image_set_path='ImageSets/Layout/val.txt',
+        pipeline=test_pipeline),
+    sampler=dict(type='DefaultSampler', shuffle=False),
+)
```
Comment (MengzhangLI) [inline configs/_base_/datasets/nih_chestxrays_bs16.py:71]:
```suggestion
test_dataloader = dict(
    batch_size=16,
    num_workers=4,
    dataset=dict(
        type=dataset_type,
        data_root='data/nih-chestxrays',
        ann_file='Data_Entry_2017.csv',
        train_test_split_file='test_list.txt',
        pipeline=test_pipeline),
    sampler=dict(type='DefaultSampler', shuffle=False),
)
```

**Comment by MengzhangLI** [comment]
Hi, @okotaku below is my result:

Model | Pretrain model | multi-label/ROCAUC
-- | -- | --
DenseNet-121 (ChexNet) | ImageNet-1k | 79.46
DenseNet-161 | ImageNet-1k | 78.76
DenseNet-169 | ImageNet-1k | 79.37
DenseNet-201 | ImageNet-1k | 78.73
ResNet-50 | ImageNet-1k | 79.96
ResNet-50 rsb | ImageNet-1k | 80.32
ResNet-101 | ImageNet-1k | 78.68
ResNet-152 | ImageNet-1k | 78.98

Noted that I use train/val split provided by https://nihcc.app.box.com/v/ChestXray-NIHCC.

File: tools/model_converters/chexnet_to_mmcls.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        state_dict = checkpoint
+
+    weight = convert_densenet(state_dict)
+    mmengine.mkdir_or_exist(osp.dirname(args.dst))
+    torch.save(dict(state_dict=weight), args.dst)
+
+    print('Done!!')
+
+
+if __name__ == '__main__':
```
Comment (MengzhangLI) [inline tools/model_converters/chexnet_to_mmcls.py:75]: I think this file is unnecessary, we can train with MMCLS provided pretrained models.

**Comment by okotaku** [comment]
@MengzhangLI The results look good. Thank you for your experiments.
